src/poolboy.erl

```diff
@@ -280,37 +277,41 @@ handle_checkin(Pid, State) ->
            waiting = Waiting,
            monitors = Monitors,
            overflow = Overflow} = State,
-    case queue:out(Waiting) of
-        {{value, {From, Ref}}, Left} ->
-            true = ets:insert(Monitors, {Pid, Ref}),
-            gen_server:reply(From, Pid),
-            State#state{waiting = Left};
-        {empty, Empty} when Overflow > 0 ->
+    case ets:first(Waiting) of
```

Comment:
I was operating based on my assumption that *the first key according to the table's internal order will be returned* (Erlang docs) meant there would order based on time of insertion, not hashed based on the waiting pid. That's unfortunate if that's how it works.